tests/scripts/regenerate_data_files_tests.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+# regenerate_data_files_tests.sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+################################################################
+#### Helpers for checking `tests/data_files`
+################################################################
+
+helper_datafile_run_tests () {
+    msg "build: $1" # ~ 1 min 50s
+    CC=gcc make
+
+    msg "test: main suites" # ~ 50s
+    CC=gcc make test
+
+    msg "test: selftest" # ~ 10s
+    programs/test/selftest
+
+    msg "test: ssl-opt.sh" # ~ 1 min
+    tests/ssl-opt.sh
+
+    msg "test: compat.sh" # ~ 6 min
+    tests/compat.sh
+
+    msg "test: context-info.sh" # ~ 15 sec
+    tests/context-info.sh
+}
+
+helper_regenerate_data_files () {
+    msg "Rebuild data files with $1"
+    msg "Build mbedtls tools"
+    make  # data_files depends on programs/x509/cert_*
+
+    msg "Cleanup final files make -C tests/data_files $1"
+    make -C tests/data_files $1
+    shift
+
+    dd if=/dev/urandom of=./tests/data_files/seedfile bs=64 count=1
+    msg "Regenerate files"
+    make -C tests/data_files all_final $*
+
+    msg "Remove intermediate files"
+    make -C tests/data_files clean
+
+    msg "Only modified files are allowd"
+    if git status -s --ignored -- tests/data_files | grep -vf tests/data_files/ignored.lst
+    then
+        err_msg "Files were not generated or not cleanup"
+        ( git status -s --ignored -- tests/data_files | grep -vf tests/data_files/ignored.lst) || true
+        err_msg "Status after regenerating"
+        git status -s --ignored -- tests/data_files
+        exit 1
+    fi
+
+    cleanup
+}
+
+component_test_regnerate_parse_input () {
+    helper_regenerate_data_files tidy all_parse_input
+}
+
+component_test_regenerate_data_files_full () {
+    helper_regenerate_data_files neat
+    helper_datafile_run_tests "default configuration with refreshed data files"
+}
+
+component_test_regenerate_data_files_default () {
+    helper_regenerate_data_files neat
+    scripts/config.py full
+    helper_datafile_run_tests "default configuration with refreshed data files"
```

Comment:
I don't think this is a good approach for checking that the new files are good. We're checking that the tests pass, but that doesn't mean, but that doesn't mean that the tests do the same thing.

I would prefer a more reliable approach: compare the new files with the old files, using a comparison method that ignores some parts of the files (content of keys, signatures and hashes), comparing only the ASN.1 structure, the metadata that isn't derived from random data (names, serial numbers, etc.), and whether signatures are correct. It will take us some time to do it, but I think the benefits are worth it.

Reply:
Concretely, [here's the general structure that I'm thinking of for the regeneration test](https://github.com/gilles-peskine-arm/mbedtls/blob/remake_test_data_files-prototype-1/tests/scripts/remake_test_data_files.py).

Reply:
Understood. I will remove them later. 

And I'd like reserve `helper_datafile_run_tests ` as local test utility. I will move it to another scripts. 

For non-ASN.1 files and malformed files, how to compare them ? Do you have any idea?

Reply:
I post those tests in this PR and reduce the scope of this PR. 

About the compare script, I'd like write it in next PR. I think the script should only compare modified files. I have add `TODO` tag in `regenerate_data_files_tests.sh` . Please check it. 

I limit openssl version to 3.0.2. I think it can generate all files. And I will fix miss or wrong commands in #7888.